src/Evolution/Systems/GrMhd/ValenciaDivClean/FiniteDifference/ReconstructWork.tpp

```diff
@@ -69,25 +69,34 @@ void compute_conservatives_for_reconstruction(
   for (size_t i = 0; i < 3; ++i) {
     spatial_velocity.get(i) /= get(lorentz_factor);
   }
+
+  // pointers to primitive variables
   const auto& rest_mass_density =
       get<hydro::Tags::RestMassDensity<DataVector>>(*vars_on_face);
   const auto& electron_fraction =
       get<hydro::Tags::ElectronFraction<DataVector>>(*vars_on_face);
-  const auto& pressure = get<hydro::Tags::Pressure<DataVector>>(*vars_on_face);
+  auto& pressure = get<hydro::Tags::Pressure<DataVector>>(*vars_on_face);
   auto& specific_internal_energy =
       get<hydro::Tags::SpecificInternalEnergy<DataVector>>(*vars_on_face);
-  auto& temperature = get<hydro::Tags::Temperature<DataVector>>(*vars_on_face);
-  if constexpr (ThermodynamicDim == 2) {
+  const auto& temperature =
+      get<hydro::Tags::Temperature<DataVector>>(*vars_on_face);
+
+  // EoS calls based on reconstructed primitives
+  if constexpr (ThermodynamicDim == 1) {
     specific_internal_energy =
-        eos.specific_internal_energy_from_density_and_pressure(
-            rest_mass_density, pressure);
-    temperature = eos.temperature_from_density_and_energy(
-        rest_mass_density, specific_internal_energy);
-  } else {
+        eos.specific_internal_energy_from_density(
+            rest_mass_density);
+    pressure = eos.pressure_from_density(rest_mass_density);
+  } else if constexpr (ThermodynamicDim == 2) {
     specific_internal_energy =
-        eos.specific_internal_energy_from_density(rest_mass_density);
-    temperature = eos.temperature_from_density(rest_mass_density);
+        eos.specific_internal_energy_from_density_and_temperature(
+            rest_mass_density, temperature);
+    pressure = eos.pressure_from_density_and_energy(rest_mass_density,
+                                                    specific_internal_energy);
+  } else if constexpr (ThermodynamicDim == 3) {
+    ERROR("1d, 2d, nor 3d EOS chosen");
```

Comment:
?  I would have expected this in a last else.   Are you trying to say 3d not implemented yet?